samza-core/src/main/java/org/apache/samza/application/TaskApplicationDescriptorImpl.java

```diff
@@ -65,6 +66,7 @@ public void addInputStream(InputDescriptor inputDescriptor) {
     // TODO: SAMZA-1841: need to add to the broadcast streams if inputDescriptor is for a broadcast stream
     Preconditions.checkState(!inputDescriptors.containsKey(inputDescriptor.getStreamId()),
         String.format("add input descriptors multiple times with the same streamId: %s", inputDescriptor.getStreamId()));
+    getOrCreateStreamSerdes(inputDescriptor.getStreamId(), inputDescriptor.getSerde());
```

Comment:
The serdes are included in the descriptor. However, a) the actual KV&lt;Serde, Serde&gt; to serialize/deserialize the incoming message key and msg may be different from the Serde in descriptor; b) there is no unified API between StreamApplicationDescriptorImpl and TaskApplicationDescriptorImpl to get the input/output serdes, since they are spread across in stream descriptors and OperatorSpec/OutputStreamImpl.